multicluster/controllers/multicluster/member/stale_controller.go

```diff
@@ -375,80 +366,156 @@ func (c *StaleResCleanupController) cleanupClusterInfoResourceExport(ctx context
 	return nil
 }
 
-func (c *StaleResCleanupController) cleanupMemberClusterAnnounces(ctx context.Context) error {
-	memberClusterAnnounceList := &mcsv1alpha1.MemberClusterAnnounceList{}
-	if err := c.List(ctx, memberClusterAnnounceList, &client.ListOptions{}); err != nil {
+// Run starts the StaleResCleanupController and blocks until stopCh is closed.
+func (c *StaleResCleanupController) Run(stopCh <-chan struct{}) {
+	klog.InfoS("Starting StaleResCleanupController")
+	defer klog.InfoS("Shutting down StaleResCleanupController")
+
+	ctx, _ := wait.ContextForChannel(stopCh)
+
+	go func() {
+		for range c.commonAreaCreationCh {
+			retry.OnError(common.CleanUpRetry, func(err error) bool { return true },
```

Comment:
I guess one issue with `retry.OnError` is that if a ClusterSet is recreated before the previous retry is done, we can have two retries ongoing the same time? But probably let us look at a better solution in a follow-up PR.